synapse/handlers/presence.py

```diff
@@ -1005,35 +1006,29 @@ async def user_syncing(
             # If they're busy then they don't stop being busy just by syncing,
             # so just update the last sync time.
             if prev_state.state != PresenceState.BUSY:
-                # XXX: We set_state separately here and just update the last_active_ts above
                 # This keeps the logic as similar as possible between the worker and single
-                # process modes. Using set_state will actually cause last_active_ts to be
-                # updated always, which is not what the spec calls for, but synapse has done
-                # this for... forever, I think.
+                # process modes.
+                # TODO: for the BUSY state, I think it would be fine to pass this in, as
+                #  it will be evaluated to update last_active_ts, which I think meets
+                #  the proposed spec change for MSC3026.
                 await self.set_state(
-                    UserID.from_string(user_id), {"presence": presence_state}, True
+                    UserID.from_string(user_id),
+                    {"presence": presence_state},
+                    ignore_status_msg=True,
                 )
                 # Retrieve the new state for the logic below. This should come from the
                 # in-memory cache.
                 prev_state = await self.current_state_for_user(user_id)
 
-            # To keep the single process behaviour consistent with worker mode, run the
-            # same logic as `update_external_syncs_row`, even though it looks weird.
-            if prev_state.state == PresenceState.OFFLINE:
-                await self._update_states(
-                    [
-                        prev_state.copy_and_replace(
-                            state=PresenceState.ONLINE,
-                            last_active_ts=self.clock.time_msec(),
-                            last_user_sync_ts=self.clock.time_msec(),
-                        )
-                    ]
-                )
```

Comment:
This code block preformed two functions, updating `last_active_ts` and `last_user_sync_ts`. `last_active_ts` is now handled by `set_state()`, and `last_user_sync_ts` will be set at the finish of the context manager, so is unneeded here.